### DISCUSSION TITLE
Set HTTP header on request to authorization endpoint

### DISCUSSION THREAD

**Discussion by mivd7** [nextauthjs/next-auth#7756] [Help]
Hi, I'm using next-auth for a custom oauth provider. I need to set a request header to the signin request as configured under the 'authorization' option. 

```
providers: [
        {
            id: process.env.CLIENT_ID,
            name: 'oauth-website',
            type: 'oauth',
            version: '2.0',
            authorization: {
                url: process.env.NEXTAUTH_URL,
                params: {
                    scope: 'user_info',
                },
            },
       //...
      }
]
```

Is there a way to configure a HTTP header that will be send along with the request to `NEXTAUTH_URL`? As the custom signin page is localized in two languages I need to provide the X-Default-Language header to direct the user to the right login page. 

I read somewhere in your documentation that I can define a request function under authorization that supposedly takes over the request handler for the authorization endpoint, which sounds like it could provide a solution for this, however this request function does not seem to get fired in any way.

**Comment by chrismuiruriz** [comment]
Hi @mivd7 did you get a solution?
